### PR TITLE
network: allow multiple IPv6 null addresses

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by bluca** [comment]
Looks new and might be related?

```
05:43:45 ======================================================================
05:43:45 ERROR: test_address_static (__main__.NetworkdNetworkTests)
05:43:45 ----------------------------------------------------------------------
05:43:45 Traceback (most recent call last):
05:43:45   File "./test/test-network/systemd-networkd-tests.py", line 2529, in test_address_static
05:43:45     ip6_null_74=ip6_null_74,
05:43:45   File "./test/test-network/systemd-networkd-tests.py", line 2449, in verify_address_static
05:43:45     prefix16 = ip4_null_16.removesuffix('.0.1')
05:43:45 AttributeError: 'str' object has no attribute 'removesuffix'
```

**Comment by yuwata** [comment]
Ah, yeah, the function is relatively new, and not supported in CentOS8. I will replace it.

**Comment by yuwata** [comment]
PTAL.

**Review by bluca — APPROVED** [review]

File: src/network/networkd-address.c [hunk 1]
```diff
@@ -394,7 +397,16 @@ static int address_compare_func(const Address *a1, const Address *a2) {
...
         case AF_INET6:
                 /* See kernel's ipv6_get_ifaddr() in net/ipv6/addrconf.c */
-                return memcmp(&a1->in_addr, &a2->in_addr, FAMILY_ADDRESS_SIZE(a1->family));
+                r = memcmp(&a1->in_addr, &a2->in_addr, FAMILY_ADDRESS_SIZE(a1->family));
+                if (r != 0)
+                        return r;
+
+                /* To distinguish IPv6 null addresses with different prefixlen, e.g. ::48 vs ::64, let's
+                 * compare the prefix length. */
+                if (a1->family == AF_INET6 && in6_addr_is_null(&a1->in_addr.in6))
```
Comment (poettering) [inline src/network/networkd-address.c]: a1-family is definitly AF_INET6 here, the switch statement already checked that. hence you can drop that part of the check here

Comment (yuwata) [reply src/network/networkd-address.c]: No,  AF_INET case has `_fallthrough_`.

**Review by keszybz — APPROVED** [review]
LGTM.

**Pull request merged**